mesh/mesh.cpp

```diff
@@ -3014,7 +3014,7 @@ void Mesh::FinalizeTetMesh(int generate_edges, int refine, bool fix_orientation)
    FinalizeCheck();
    CheckElementOrientation(fix_orientation);
 
-   if (NumOfBdrElements == 0)
+   if (ReduceInt(NumOfBdrElements) == 0)
```

Comment:
Can we work around this? I don't like introducing parallel paradigms into serial classes

Reply:
Welcome to MFEM 😄 This is the way how compatibility is achieved between `Mesh` and `ParMesh`, so `Mesh::ReduceInt` does nothing in fact 🙃 . I do not think there is a real work around, maybe cosmetic, but global number of (boundary) elements is not stored, which would require an update every time something happens somewhere globally, which is error prone 💣 . Instead, the other approach is taken collecting the global numbers when you need them, which should be (and normally is) rare.

Reply:
Just because there are occurrences in the codebase where suboptimal choices have been made doesn't mean we ignore better practice in the future.

Why can the reduce not happen at the `ParMesh::GenerateBoundaryElements` level?

Reply:
And then someone tells me to keep the changes minimal and why do I make things complicated 😞 ... but that is a different story, I definitely support cleaning the code bit by bit 😉 
The thing here is that `GenerateBoundaryElements` does not check anything and overwrites the existing boundary eventually, which might be desired behavior, so it should not go there. However, we may introduce something like `EnsureBoundaryElements` (like `EnsureParNodes` or other), which will do the check and call `GenerateBoundaryElements` if needed. It may then have serial and parallel implementations (i.e. `Mesh` vs. `ParMesh`). How about that? 😉

Reply:
Unfortunately, that will not help everywhere, because more things are done at some places. The only thing I see as possible is to add `GetGlobalNBE` like the existing `GetGlobalNE` 🤔 .

Reply:
...or use `HasBoundaryElements()` 🤔

Reply:
Ok, here we go, `HasBoundaryElements()` sounds as the cleanest way 👍 . Interestingly, it was not used anywhere 🙃

Reply:
ping @jandrej